cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -194,7 +195,11 @@ public BodyConsumer create(HttpRequest request, HttpResponder responder,
     String versionId = ApplicationId.DEFAULT_VERSION;
     // If LCM flow is enabled - we generate specific versions of the app.
     if (Feature.LIFECYCLE_MANAGEMENT_EDIT.isEnabled(featureFlagsProvider)) {
-      versionId = RunIds.generate().getId();
+      // To solve the concurrent deployment issue, we use the
+      // same 0 time-based UUID as the first application version
+      versionId = applicationLifecycleService.applicationExists(new ApplicationReference(namespaceId, appId))
+        ? RunIds.generate(0).getId()
```

Comment:
Two calls to this will generate different ids (if it didn't, there would be no point in this as it would be functionally the same as using the timestamp). Only the time part of the UUID will be the same.

There needs to be a globally static value for the first version, like '1'.